gobblin-metrics/src/test/java/gobblin/metrics/performance/PerformanceUtils.java

```diff
@@ -0,0 +1,204 @@
+   * @param forkAtDepth If multiple threads, each thread has its own {@link gobblin.metrics.MetricContext}. This
+   *                    parameter sets the first level in the tree where the per-thread MetricContexts branch off.
+   * @param counters Number of counters to generate per thread.
+   * @param meters Number of meters to generate per thread.
+   * @param histograms Number of histograms to generate per thread.
+   * @param timers Number of timers to generate per thread.
+   * @param queries Number of increments to do, divided among all threads.
+   * @throws Exception
+   */
+  @Builder(buildMethodName = "run", builderMethodName = "multiTest")
+  public static void _multiTest(@Singular("threads") Set<Integer> threads,
+      @Singular("depth") Set<Integer> depth,
+      @Singular("forkAtDepth") Set<Integer> forkAtDepth,
+      @Singular("counters") Set<Integer> counters,
+      @Singular("meters") Set<Integer> meters,
+      @Singular("histograms") Set<Integer> histograms,
+      @Singular("timers") Set<Integer> timers,
+      long queries, String name) throws Exception {
+
+    if(threads.isEmpty()) {
+      threads = Sets.newHashSet(1);
+    }
+    if(forkAtDepth.isEmpty()) {
+      forkAtDepth = Sets.newHashSet(0);
+    }
+    if(depth.isEmpty()) {
+      depth = Sets.newHashSet(0);
+    }
+    if(counters.isEmpty()) {
+      counters = Sets.newHashSet(0);
+    }
+    if(meters.isEmpty()) {
+      meters = Sets.newHashSet(0);
+    }
+    if(histograms.isEmpty()) {
+      histograms = Sets.newHashSet(0);
+    }
+    if(timers.isEmpty()) {
+      timers = Sets.newHashSet(0);
+    }
+    if(queries == 0) {
+      queries = 50000000l;
+    }
+    if(Strings.isNullOrEmpty(name)) {
+      name = "Test";
+    }
+
+    Set<List<Integer>> parameters = Sets.cartesianProduct(threads, depth, forkAtDepth, counters, meters,
+        histograms, timers);
+
+    Comparator<List<Integer>> comparator = new Comparator<List<Integer>>() {
+      @Override public int compare(List<Integer> o1, List<Integer> o2) {
+        Iterator<Integer> it1 = o1.iterator();
+        Iterator<Integer> it2 = o2.iterator();
+
+        while(it1.hasNext() && it2.hasNext()) {
+          int compare = Integer.compare(it1.next(), it2.next());
+          if(compare != 0) {
+            return compare;
+          }
+        }
+        if(it1.hasNext()) {
+          return 1;
+        } else if(it2.hasNext()) {
+          return -1;
+        } else {
+          return 0;
+        }
+      }
+    };
+
+    TreeMap<List<Integer>, Double> results = Maps.newTreeMap(comparator);
+
+    for(List<Integer> p : parameters) {
+      results.put(p, _singleTest(p.get(0), p.get(1), p.get(2), p.get(3), p.get(4), p.get(5), p.get(6), queries));
+    }
+
+    System.out.println("===========================");
+    System.out.println(name);
+    System.out.println("===========================");
+    System.out.println("Threads\tDepth\tForkAtDepth\tCounters\tMeters\tHistograms\tTimers\tQPS");
```

Comment:
Because print statements are automatically captured by TestNG and put into the testing report, while log statements aren't.